src/test/java/org/dependencytrack/resources/v1/ComponentResourceTest.java

```diff
@@ -77,71 +84,103 @@ public void getOutdatedComponentsTest() throws MalformedPackageURLException {
             component.setVersion(String.valueOf(i)+".0");
             component.setPurl(new PackageURL(RepositoryType.MAVEN.toString(), "component-group", "component-name-"+i , String.valueOf(i)+".0", null, null));
             component = qm.createComponent(component, false);
-            if (i<100) {
-                if ((i >= 25) && (i < 75)) {
-                    // 50 recent direct depencencies
-                    directDepencencies.add("{\"uuid\":\"" + component.getUuid() + "\"}");
-                }
-                // same version
+            // direct depencencies
+            if (i < 100) {
+                // 100 direct depencencies, 900 transitive depencencies
+                directDepencencies.add("{\"uuid\":\"" + component.getUuid() + "\"}");
+            }
+            // Recent & Outdated
+            if ((i >= 25) && (i < 225)) {
+                // 100 outdated components, 75 of these are direct dependencies, 25 transitive
                 final var metaComponent = new RepositoryMetaComponent();
                 metaComponent.setRepositoryType(RepositoryType.MAVEN);
                 metaComponent.setNamespace("component-group");
                 metaComponent.setName("component-name-"+i);
-                metaComponent.setLatestVersion(String.valueOf(i)+".0");
+                metaComponent.setLatestVersion(String.valueOf(i+1)+".0");
                 metaComponent.setLastCheck(new Date());
                 qm.persist(metaComponent);
-            }
-            if (i>=100 && i<200) {
-                if ((i >= 150) && (i < 175)) {
-                    // 25 outdated direct depencencies
-                    directDepencencies.add("{\"uuid\":\"" + component.getUuid() + "\"}");
-                }
-                // newer version
+            } else if (i<500) {
+                // 300 recent components, 25 of these are direct dependencies
                 final var metaComponent = new RepositoryMetaComponent();
                 metaComponent.setRepositoryType(RepositoryType.MAVEN);
                 metaComponent.setNamespace("component-group");
                 metaComponent.setName("component-name-"+i);
-                metaComponent.setLatestVersion(String.valueOf(i+1)+".0");
+                metaComponent.setLatestVersion(String.valueOf(i)+".0");
                 metaComponent.setLastCheck(new Date());
                 qm.persist(metaComponent);
+            } else {
+                // 500 components with no version metadata, all transitive dependencies
```

Comment:
@walterdeboer Is the test here trying to state that if there is no metadata provided in the SBOM, or if a manually created component is made without metadata, that we assume it is transitive?  Just trying to understand this comment here.

Reply:
This has nothing to do with the SBOM, when DT detects new versions in the remote repository, it creates a RepositoryMetaComponent. The components here have no RepositoryMetaComponent, so they represent te most current versions